### ISSUE TITLE
[❓Question] 自定义切分数据集，但未能成功见到评估

### ISSUE THREAD

**Issue by vincentForWss** [RUCAIBox/RecBole#1110]
您好，我参考了文档中自定义数据集的方法，方法如下：
## Benchmark file
benchmark_filename (list) : List of pre-split user-item interaction suffix. We will only apply normalize, remap-id, which will not delete the interaction in inter_feat. And then split the inter_feat by benchmark_filename. E.g. Let’s assume that the dataset is called click, and benchmark_filename equals to ['part1', 'part2', 'part3']. That we will load click.part1.inter, click.part2.inter, click.part3.inter, and treat them as train, valid, test dataset. Defaults to None.

但我在运行代码时，发现原来的验证Evaluate不再报告在输出界面了，想问下是否有一系列相应的配置需求才能输出报告？
运行如下：
Train    16: 100%|██████████████████████████| 17/17 [00:02<00:00,  7.11it/s, GPU RAM: 2.84 G/6.00 G]
07 Jan 13:14    INFO  epoch 16 training [time: 2.39s, train loss: 0.1654]
Evaluate   : 100%|████████████████████████████| 3/3 [00:00<00:00, 21.17it/s, GPU RAM: 2.84 G/6.00 G]
07 Jan 13:14    INFO  epoch 16 evaluating [time: 0.30s, valid_score: 0.202900]
07 Jan 13:14    INFO  valid result: 
auc : 0.8495    logloss : 0.1238    f2 : 0.2029

**Comment by Sherry-XLL** [comment]
@vincentForWss 你好，自定义数据集只是改变了读入数据的方式，与评测输出没有直接关系。 `verbose` 决定是否输出训练损失和验证的各指标，默认情况下设置为 `True`， 参数 `show_progress (bool)` 决定是否显示训练和验证的 epochs，你可以在 `trainer` 部分找到相关的代码。

https://github.com/RUCAIBox/RecBole/blob/4bfcf336c7cd8e918936118ae6bbca7fba121fe5/recbole/trainer/trainer.py#L348-L350

如果你在直接读入数据时不出现该问题，只是使用 `benchmark_file` 时才缺少验证的输出，可能是 `valid_data` 没有正确加载，你可以通过调试输出查看相关结果。

**Comment by vincentForWss** [comment]
是这个的问题"如果你在直接读入数据时不出现该问题，只是使用&nbsp;benchmark_file&nbsp;时才缺少验证的输出，可能是&nbsp;valid_data&nbsp;没有正确加载，你可以通过调试输出查看相关结果。"我这边需要做什么操作才能将part2识别为验证集呢？

目前报告如下：
Trainable parameters: 12294613
2022-01-07 13:53:16.926395: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library cudart64_110.dll
Train&nbsp; &nbsp; &nbsp;0: 100%|██████████████████████████| 24/24 [00:04<00:00,&nbsp; 5.84it/s, GPU RAM: 3.04 G/6.00 G]
07 Jan 13:53&nbsp; &nbsp; INFO&nbsp; epoch 0 training [time: 4.11s, train loss: 2.3805]
Train&nbsp; &nbsp; &nbsp;1: 100%|██████████████████████████| 24/24 [00:03<00:00,&nbsp; 6.88it/s, GPU RAM: 3.04 G/6.00 G]
07 Jan 13:53&nbsp; &nbsp; INFO&nbsp; epoch 1 training [time: 3.49s, train loss: 0.8298]
07 Jan 13:53&nbsp; &nbsp; INFO&nbsp; best valid : None
07 Jan 13:53&nbsp; &nbsp; INFO&nbsp; test result: None

目录结构如下：








------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Lanling ***@***.***&gt;; 
发送时间: 2022年1月7日(星期五) 中午1:49
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [RUCAIBox/RecBole] [❓Question] 自定义切分数据集，但未能成功见到评估 (Issue #1110)





 
@vincentForWss 你好，自定义数据集只是改变了读入数据的方式，与评测输出没有直接关系。 verbose 决定是否输出训练损失和验证的各指标，默认情况下设置为 True， 参数 show_progress (bool) 决定是否显示训练和验证的 epochs，你可以在 trainer 部分找到相关的代码。
 
https://github.com/RUCAIBox/RecBole/blob/4bfcf336c7cd8e918936118ae6bbca7fba121fe5/recbole/trainer/trainer.py#L348-L350
 
如果你在直接读入数据时不出现该问题，只是使用 benchmark_file 时才缺少验证的输出，可能是 valid_data 没有正确加载，你可以通过调试输出查看相关结果。
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by vincentForWss** [comment]
dataset
- data
- - data.part1.inter
- - data.part2.inter
- - data.part3.inter

**Comment by Sherry-XLL** [comment]
@vincentForWss 我用 `benchmark_file` 试了一下同样的配置，输出是正常的。根据你的输出，你的 `valid_data` 和 `test_data` 没有正确加载，使得验证集和测试集都为空，你可以通过 `log` 检查配置信息中 `benchmark_filename` 是否正确读入以及数据集是否是 `data`。

**Comment by vincentForWss** [comment]
可以确信数据集是data，benchmark_filename正确读入
关于eval.yaml
`# model config
embedding_size: 4

eval_args:
  group_by: ~
  order: TO
#  split:
##    LS: valid_and_test
#    RS: [0.95,0.025,0.025]
  mode: labeled

training_neg_sample_num: 0 #负采样数目
eval_step: 1 #每次训练后做evalaution的次数
# evalution settings
#eval_setting: RO_RS #对数据随机重排，设置按比例划分数据集
#group_by_user: False #是否将一个user的记录划到一个组里
#split_ratio: [ 0.8,0.1,0.1 ] #切分比例
metrics: [ 'AUC', 'LogLoss','f2' ] #评测标准
train_metric: f2
valid_metric: f2 #选取哪个评测标准作为作为提前停止训练的标准
eval_batch_size: 1024 #评测的batch_size
`
以下是输出日志
`General Hyper Parameters:
gpu_id = 0
use_gpu = True
seed = 2021
state = INFO
reproducibility = True
data_path = dataset/data
show_progress = True
save_dataset = False
save_dataloaders = False
benchmark_filename = ['part1', 'part2', 'part3']

Training Hyper Parameters:
checkpoint_dir = saved
epochs = 2
train_batch_size = 4096
learner = adam
learning_rate = 0.01
eval_step = 1
stopping_step = 10
clip_grad_norm = None
weight_decay = 0.0
loss_decimal_place = 4
training_neg_sample_num = 0

Evaluation Hyper Parameters:
eval_args = {'group_by': None, 'order': 'TO', 'mode': 'labeled', 'split': {'RS': [0.8, 0.1, 0.1]}}
metrics = ['AUC', 'LogLoss', 'f2']
topk = [10]
valid_metric = f2
valid_metric_bigger = True
eval_batch_size = 1024
metric_decimal_place = 4

Dataset Hyper Parameters:
field_separator = 	
seq_separator =  
USER_ID_FIELD = user_id
ITEM_ID_FIELD = item_id
RATING_FIELD = rating
TIME_FIELD = timestamp
seq_len = None
LABEL_FIELD = label
threshold = {'rating': 1}
NEG_PREFIX = neg_
load_col = {'inter': ['user_id', 'item_id', 'rating', 'timestamp', '净现金收入', '累计现金收入'], 'user': ['user_id', '客户性别', '客户等级', '客户年龄', '客户最新风险等级', '客户风险向好性努力程度', '客户风险等级评估距今总天数'], 'item': ['item_id', '理财产品计价类型', '理财产品周期类型', '理财产品产品模式', '理财产品风险等级', '理财产品是否滚存型产品', '理财产品是否允许变更分红方式', '理财产品持有天数', '理财产品募集方式', '理财产品管理方式', '理财产品业务模式', '理财产品收益特点', '理财产品期限', '理财产品投资模式', '理财产品预期收益率', '理财产品最高收益率', '理财产品最低收益率', '理财产品展示等级', '理财产品类型', '理财产品产品模式2', '理财产品收益记息基数', '理财产品预期最低收益率', '理财产品预期最高收益率', '理财产品产品面值', '理财产品发行价格', '理财产品收益率', '理财产品距今总天数']}
unload_col = None
unused_col = {'inter': ['timestamp'], 'item': ['理财产品距今总天数']}
additional_feat_suffix = None
rm_dup_inter = last
val_interval = None
filter_inter_by_user_or_item = True
user_inter_num_interval = [0,inf)
item_inter_num_interval = [0,inf)
alias_of_user_id = None
alias_of_item_id = None
alias_of_entity_id = None
alias_of_relation_id = None
preload_weight = None
normalize_field = None
normalize_all = True
ITEM_LIST_LENGTH_FIELD = item_length
LIST_SUFFIX = _list
MAX_ITEM_LIST_LENGTH = 50
POSITION_FIELD = position_id
HEAD_ENTITY_ID_FIELD = head_id
TAIL_ENTITY_ID_FIELD = tail_id
RELATION_ID_FIELD = relation_id
ENTITY_ID_FIELD = entity_id

Other Hyper Parameters: 
neg_sampling = None
repeatable = False
embedding_size = 4
mlp_hidden_size = [128, 128, 128]
dropout_prob = 0.2
MODEL_TYPE = ModelType.CONTEXT
encoding = utf-8
train_metric = f2
MODEL_INPUT_TYPE = InputType.POINTWISE
eval_type = EvaluatorType.VALUE
device = cuda
train_neg_sample_args = {'strategy': 'none'}
eval_neg_sample_args = {'strategy': 'none', 'distribution': 'none'}`

**Comment by Sherry-XLL** [comment]
@vincentForWss 你好，我使用 `ml-1m` 数据集，按照如下配置文件运行 context model 里的 `FM` 模型输出也是正常的，你可以试着换成其他数据集（比如 `ml-1m`）看看问题是否依然存在。
```python
python run_recbole.py --model=FM --dataset=ml-1m --config_files=ml-1m_context.yaml
```

```
# dataset config
field_separator: "\t"
seq_separator: " "
USER_ID_FIELD: user_id
ITEM_ID_FIELD: item_id
RATING_FIELD: rating
NEG_PREFIX: neg_
LABEL_FIELD: label
threshold:
    rating: 4
load_col:
    inter: [user_id, item_id, rating, timestamp]
    user: [user_id, age, gender, occupation]
    item: [item_id, genre]
unused_col:
    inter: [timestamp]

neg_sampling: ~

# training and evaluation
epochs: 500
train_batch_size: 4096
eval_batch_size: 40960000
eval_args:
  split: {'RS':[0.8, 0.1, 0.1]}
  group_by: ~
  mode: labeled
  order: TO
valid_metric: AUC
metrics: ['AUC', 'LogLoss']
benchmark_filename: ['part1', 'part2', 'part3']

# model
embedding_size: 10
```
